sklearn/metrics/_pairwise_distances_reduction/_dispatcher.py

```diff
@@ -96,7 +96,9 @@ def is_usable_for(cls, X, Y, metric) -> bool:
         """
 
         def is_numpy_c_ordered(X):
```

Comment:
Side-note: if there are any other libraries that implement the buffer protocol in Python and that we could support, we might want to adapt this.